pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -40,8 +40,10 @@ import (
 var redisOpTimeout = 5 * time.Second
 
 // buildInboundNetworkFilters generates a TCP proxy network filter on the inbound path
-func buildInboundNetworkFilters(push *model.PushContext, instance *model.ServiceInstance, node *model.Proxy) []*listener.Filter {
-	clusterName := model.BuildInboundSubsetKey(int(instance.Endpoint.EndpointPort))
+func buildInboundNetworkFilters(push *model.PushContext, instance *model.ServiceInstance, node *model.Proxy, clusterName string) []*listener.Filter {
+	if clusterName == "" {
```

Comment:
When will cluster name be empty?

Reply:
for the per-port filter chain we pass it empty. Not really sure why, I was copying the pattern we use elsewhere. I can make. it compute it before calling the function

Reply:
Yeah. That is better to avoid special handling unless it throws some surprises :-)